sql/src/main/java/io/crate/analyze/DeallocateAnalyzer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.analyze.expressions.ExpressionToStringVisitor;
+import io.crate.sql.tree.DeallocateStatement;
+
+public final class DeallocateAnalyzer {
+
+    private DeallocateAnalyzer() {
+    }
+
+    public static DeallocateAnalyzedStatement analyze(DeallocateStatement deallocateStatement,
+                                                      ParameterContext parameterContext) {
+        if (deallocateStatement.preparedStmt() != null) {
+            return new DeallocateAnalyzedStatement(ExpressionToStringVisitor
```

Comment:
This is not compatible with the unbound analysis which we should support for all new things. The `DeallocateAnalyzedStatement` should also implement `visitSymbols` to support type-inference for the postgres protocol.

(Or we should drop the placeholder/param support from the parser for DEALLOCATE)

Reply:
I think we should drop the placeholder/param support... I don't see it being used like that (postgres and coackroach)